tokenizer_functions.py

```diff
@@ -0,0 +1,85 @@
+import logging
+import re
+from collections import Counter
+
+# Configuration logging
+logging.basicConfig(level=logging.DEBUG, format='%(levelname)s:%(message)s')
+
+def clean_text(input_text):
+    """
+    Convert input text to lowercase and remove punctuation.
+    
+    Params
+    ----------
+    input_text : str
+        The string to clean.
+        
+    Returns
+    -------
+    str
+        The cleaned string.
+    
+    Raises
+    ------
+    AssertionError
+        If the input is not a string or if the output is not a string.
+    """
+    assert isinstance(input_text, str), "Input must be a string"
```

Comment:
Awesome that you have comments to help a fellow coder!

And you have asserts, cool!

Only thing here is maybe some spacing to make things easier to digest.  For example

```python
    assert isinstance(input_text, str), "Input must be a string"

    logging.debug("Cleaning text: %s", input_text)
    cleaned_text = re.sub(r'[^\w\s]', '', input_text.lower())
    logging.debug("Cleaned text: %s", cleaned_text)

    assert isinstance(cleaned_text, str), "Output must be a string"
    return cleaned_text
```

When I see that I can more easily see that the assert is part of a guard on the input, the middle is where heart of the function is processed and has logging, and finally the last assert belongs to the return.

All good and working, just looking at readabilty down the line.